runtime/src/lib.rs

```diff
@@ -214,11 +214,22 @@ parameter_types! {
 	pub const SS58Prefix: u8 = 13;
 }
 
+//Don't allow any token actions
+pub struct BaseFilter;
+impl Contains<Call> for BaseFilter {
+	fn contains(call: &Call) -> bool {
+		!matches!(call, Call::Balances(..) | Call::Treasury(..) | Call::Vesting(_))
+	}
+}
+
 // Configure FRAME pallets to include in runtime.
 
 impl frame_system::Config for Runtime {
 	/// The basic call filter to use in dispatchable.
+	#[cfg(not(feature = "mainnet-launch"))]
 	type BaseCallFilter = frame_support::traits::Everything;
```

Comment:
while we're on it, we could solve https://github.com/integritee-network/pallet-teerex/issues/46 too

In that sense, there will only be BaseFilter here and the compiler flag would rather be inside the BaseFilter definition

Reply:
and please make a comment in the code why we filter it: https://github.com/integritee-network/pallet-teerex/issues/7